ibis/expr/types/strings.py

```diff
@@ -93,7 +93,7 @@ def __getitem__(self, key: slice | int | ir.IntegerScalar) -> StringValue:
         if isinstance(key, slice):
             start, stop, step = key.start, key.stop, key.step
 
-            if step is not None and not isinstance(step, ir.Expr) and step != 1:
+            if step is not None and step != 1:
```

Comment:
Is this raising a `ValueError` when `step` is an Ibis expression because of the override of `__bool__` on `Expr`?

Reply:
Oh, oops this is raising a `ValueError: The truth value of an Ibis expression is not defined` and the test is actually catching that. I want it to actually catch this custom error. fixing now.